pkg/util/containers/metrics/system/containerid_linux.go

```diff
@@ -57,3 +78,40 @@ func parseMountinfo(path string) (string, error) {
 
 	return "", nil
 }
+
+// parseCgroupNodePath parses /proc/self/cgroup and returns the path to the cgroup node controller.
+func parseCgroupNodePath(r io.Reader, controller string) string {
```

Comment:
why do we repeat again the logic parsing logic in another package? 
can we have a common implementation?

Reply:
If we can't directly use the other package due to dependency conflict, can we create a new package to share the logic?